pkg/github/repositories.go

```diff
@@ -471,7 +488,39 @@ func GetFileContents(getClient GetClientFn, t translations.TranslationHelperFunc
 
 			var result interface{}
 			if fileContent != nil {
-				result = fileContent
+				if fileContent.Content != nil && (begin > 0 || end > 0) {
+					decoded, err := fileContent.GetContent()
+					if err != nil {
+						return nil, fmt.Errorf("failed to decode file content: %w", err)
+					}
+					lines := strings.Split(decoded, "\n")
+					totalLines := len(lines)
+					// Adjust indices for 1-based input
+					startIdx := begin - 1
+					if startIdx < 0 {
+						startIdx = 0
+					}
+					endIdx := end
+					if endIdx <= 0 || endIdx > totalLines {
+						endIdx = totalLines
+					}
+					if startIdx >= totalLines {
+						startIdx = totalLines - 1
+					}
+					if startIdx < 0 {
+						startIdx = 0
+					}
+					if endIdx < startIdx {
+						endIdx = startIdx
+					}
+					ranged := lines[startIdx:endIdx]
+					joined := strings.Join(ranged, "\n")
+					result = &github.RepositoryContent{
```

Comment:
It seems like we should check the `.Encoding` of the file content before deciding that we need to base64 encode it.

Reply:
Fix